streamz/dataframe/core.py

```diff
@@ -314,7 +327,16 @@ class Series(Frame, _SeriesMixin):
     --------
     DataFrame
     """
-    _subtype = pd.Series
+    _subtype = object
+
+    def __init__(self, *args, **kwargs):
+        example = kwargs.get('example', None) if "example" in kwargs else args[1]
+        assert example is not None
+        if not is_series_like(example):
+            msg = "Streaming Series expects an example of Series like objects. Got: {}.".format(example)
+            raise TypeError(msg)
+        self._subtype = type(example)
+        super(Series, self).__init__(*args, **kwargs)
```

Comment:
Maybe we can include the predicate like `is_series_like` on the class and then keep this constructor on a parent class in order to avoid duplication?

Reply:
But it wouldn't be possible to set _subtype without the constructor. Also, the parent class is Streaming which should not have DataFrame specific code.

Reply:
I have used enums in an attempt to get rid of code duplication. Would love to hear your thoughts on this latest approach.